jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java

```diff
@@ -427,9 +430,54 @@ public boolean isOffline() {
    */
   @VisibleForTesting
   Path getJarArtifact() {
-    // TODO: use maven-jar-plugin's <outputDirectory> and <classifier> (i.e.,
-    // "<outputDirectory>/<finalName>-<classifier>.jar").
-    String jarName = project.getBuild().getFinalName() + ".jar";
-    return Paths.get(project.getBuild().getDirectory(), jarName);
+    String classifier = null;
+    Path buildDirectory = Paths.get(project.getBuild().getDirectory());
+    Path outputDirectory = buildDirectory;
+
+    // Read <classifier> and <outputDirectory> from maven-jar-plugin.
+    Plugin jarPlugin = project.getPlugin("org.apache.maven.plugins:maven-jar-plugin");
+    if (jarPlugin != null) {
+      for (PluginExecution execution : jarPlugin.getExecutions()) {
+        if ("default-jar".equals(execution.getId())) {
+          Xpp3Dom configuration = (Xpp3Dom) execution.getConfiguration();
+          classifier = getChildValue(configuration, "classifier").orElse(null);
+          Optional<String> directoryString = getChildValue(configuration, "outputDirectory");
+
+          if (directoryString.isPresent()) {
+            outputDirectory = project.getBasedir().toPath().resolve(directoryString.get());
+          }
+        }
+      }
+    }
+
+    String suffix = ".jar";
+    if (jarRepackagedBySpringBoot()) {
+      consoleLogger.log(
+          Level.LIFECYCLE, "Spring Boot repackaging (fat JAR) detected; using the original JAR");
+      if (outputDirectory.equals(buildDirectory)) { // Spring renames original only when needed
+        suffix += ".original";
+      }
+    }
+
+    String jarName =
+        project.getBuild().getFinalName() + (classifier == null ? "" : '-' + classifier) + suffix;
+    consoleLogger.log(Level.LIFECYCLE, "Using JAR: " + outputDirectory.resolve(jarName));
```

Comment:
Does this need to be logged at `lifecycle`? Would be nice to keep log output minimal.